dvc/scm/git.py

```diff
@@ -89,10 +90,24 @@ def __init__(self, root_dir=os.curdir, search_parent_directories=True):
         self.ignored_paths = []
         self.files_to_track = set()
 
+        self._dulwich_repo = None
+
+    @property
+    def dulwich_repo(self):
```

Comment:
Thank you so much for using dulwich right away! :pray: Is the functionality sufficient?

Also, why not cached_property?

Reply:
For the refs related features dulwich will work for us. We just have to make sure to do the necessary conversion from strings to bytes when making dulwich calls.

The dulwich stash implementation is mostly incomplete, but the parts of it which are implemented work well enough.

And I used property here because we want to call `_dulwich_repo.close()` in `scm.close()`, but we won't always be instantiating the dulwich repo